### ISSUE TITLE
Missing keyboard focus when opening Mod Select during startup lag spikes

### ISSUE THREAD

**Issue by Wilker-uwu** [ppy/osu#38682]
### Type

Game behaviour

### Bug description

using shortcut keys for mod selection during the initial loading of the game gets it to skip the screen from grabbing keyboard focus, so other keys pressed does song selection actions. this _may_ be dependent on the lag spikes to reproduce, it is consistent on my pc because it is slow.

the exact sequence i did for the screenshots, from right at the startup of the game is `Enter`, `Enter`, `Enter`, `F1`, `f` `l`, `Enter`, `Escape`, (selection of the Flashlight mod using the mouse click and cursor), `Escape`, with screenshots taken after the last Enter press.

### Screenshots or videos

<img width="1920" height="1080" alt="Screenshot of the game, the mod menu is open, but in the background, the song loading screen is up and running." src="https://github.com/user-attachments/assets/72d064ec-89ed-48bf-95af-fdeb4502da6b" />
pressing Enter after opening the menu will select the song instead of selecting a mod. keyboard focus is there, but it didn't seem to receive any typing.  
(i don't know why the difficulty increase mods aren't showing up in this screenshot, i didn't notice until i started writing this issue, this may be a different thing altogether)
 
<img width="1920" height="1080" alt="Screenshot of the game, the mod menu is open, but after pressing Escape from the loading screen, the Game Start button from the song selection screen is in the bottom right of the screen, drawn over the mod menu." src="https://github.com/user-attachments/assets/c43fe857-a279-4714-9b6e-3354299c5ccb" />
pressing Escape after opening the selected song returns from the loading screen back to the song selection instead of closing the menu, and the osu! logo that starts the game is drawn over the menu instead of below.  
there are also none of the usual items about mod information that shows up in the bottom bar.

<img width="1920" height="1080" alt="Screenshot of the game, the main menu retracting back to the osu! logo mid-animation" src="https://github.com/user-attachments/assets/20c6b2a7-dc90-4481-9cdb-455f69119b1e" />
pressing Escape again will take you back to the main menu. the retraction animation that happens when being idle does automatically close the mod selection menu, but it will also deselect any mods that were selected using the mouse.

### Version

2026.821.0-tachyon

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/31341030/compressed-logs.zip)

**Comment by Wilker-uwu** [comment]
the F2 button for random song selection also seems to stop responding after producing the bug
